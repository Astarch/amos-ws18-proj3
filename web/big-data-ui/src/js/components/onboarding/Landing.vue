<template>
  <div id="landing-page">

    <landing-header
      v-on:openRegistration="open(FormTypeEnum.register)"
      v-on:openLogin="open(FormTypeEnum.login)">
      <git-info :isActive="isGitInfoShown"></git-info>
    </landing-header>
    <!-- Header -->
    <a name="about"></a>
    <div class="intro-header" v-bind:style="{'padding-top': isGitInfoShown ? '80px': '50px'}">
      <div class="container">

        <div class="row">
          <div class="col-lg-12">
            <div class="intro-logo">
              <img class="img-fluid" src="~assets/logo/logo.png">
            </div>
            <div class="intro-message">
              <h1>pretrendr</h1>
              <h3>Analyse and Visualize your data and identify trends</h3>
              <hr class="intro-divider">
              <ul class="list-inline intro-social-buttons">
                <li>
                  <a href="#register" ref="register"
                     class="btn btn-default btn-lg"
                     v-on:click.prevent="open(FormTypeEnum.register)">Register Now</a>
                </li>
                <li>
                  <a href="#login" class="btn btn-default btn-lg"
                     v-on:click.prevent="open(FormTypeEnum.login)">Login</a>
                </li>
              </ul>
            </div>
          </div>
        </div>

      </div>
      <!-- /.container -->
    </div>
    <!-- /.intro-header -->

    <!-- Page Content -->
    <a name="services"></a>
    <div class="content-section-a">

      <div class="container">
        <div class="row">
          <div class="col-lg-5 col-sm-6">
            <hr class="section-heading-spacer">
            <div class="clearfix"></div>
            <h2 class="section-heading">Death to the Stock Photo:<br>Special Thanks</h2>
            <p class="lead">A special thanks to <a target="_blank" href="http://join.deathtothestockphoto.com/">Death to the Stock Photo</a>
              for providing the photographs that you see in this template. Visit their website to become a member.</p>
          </div>
          <div class="col-lg-5 col-lg-offset-2 col-sm-6">
            <img class="img-responsive" src="~assets/landing-page/ipad.png" alt="">
          </div>
        </div>

      </div>
      <!-- /.container -->

    </div>
    <!-- /.content-section-a -->

    <div class="content-section-b">

      <div class="container">

        <div class="row">
          <div class="col-lg-5 col-lg-offset-1 col-sm-push-6  col-sm-6">
            <hr class="section-heading-spacer">
            <div class="clearfix"></div>
            <h2 class="section-heading">3D Device Mockups<br>by PSDCovers</h2>
            <p class="lead">
              Turn your 2D designs into high quality, 3D product shots in seconds using free Photoshop actions by <a
              target="_blank" href="http://www.psdcovers.com/">PSDCovers</a>! Visit their website to download some of their awesome, free photoshop actions!
            </p>
          </div>
          <div class="col-lg-5 col-sm-pull-6  col-sm-6">
            <img class="img-responsive" src="~assets/landing-page/dog.png" alt="">
          </div>
        </div>

      </div>
      <!-- /.container -->

    </div>
    <!-- /.content-section-b -->

    <div class="content-section-a">

      <div class="container">

        <div class="row">
          <div class="col-lg-5 col-sm-6">
            <hr class="section-heading-spacer">
            <div class="clearfix"></div>
            <h2 class="section-heading">Google Web Fonts and<br>Font Awesome Icons</h2>
            <p class="lead">This template features the 'Lato' font, part of the <a target="_blank"
                                                                                   href="http://www.google.com/fonts">Google Web Font library</a>, as well as <a
              target="_blank" href="http://fontawesome.io">icons from Font Awesome</a>.</p>
          </div>
          <div class="col-lg-5 col-lg-offset-2 col-sm-6">
            <img class="img-responsive" src="~assets/landing-page/phones.png" alt="">
          </div>
        </div>

      </div>
      <!-- /.container -->

    </div>
    <!-- /.content-section-a -->

    <a name="contact"></a>
    <div class="banner">

      <div class="container">

        <div class="row">
          <div class="col-lg-6">
            <h2>Connect to pretrendr:</h2>
          </div>
          <div class="col-lg-6">
            <ul class="list-inline banner-social-buttons">
              <li>
                <a href="https://github.com/Astarch/amos-ws18-proj3" class="btn btn-default btn-lg"><i
                  class="fa fa-github fa-fw"></i> <span class="network-name">Github</span></a>
              </li>
            </ul>
          </div>
        </div>

      </div>
      <!-- /.container -->

    </div>
    <!-- /.banner -->

    <landing-footer></landing-footer>

    <onboarding-modal
      :is-active="modalActive"
      :type="modalType"
      v-on:close="modalActive = false"
      v-on:changeType="newType => modalType = newType"></onboarding-modal>


  </div>
</template>

<script>
  import GitInfo from './../../components/common/GitInfo';
  import OnboardingModal from './modal/OnboardingModal';
  import LandingHeader from './LandingHeader';
  import LandingFooter from './LandingFooter';

  import {FormTypeEnum} from './../../utils/constants';

  export default {
    name: 'Landing',
    components: {
      GitInfo, OnboardingModal, LandingHeader, LandingFooter
    },
    data: () => ({
      modalActive: false,
      modalType: FormTypeEnum.login,
      requestAnswer: '',
      FormTypeEnum
    }),
    methods: {
      open(which) {
        this.modalActive = true;
        this.modalType = which;
      },
      close(event) {
        event.preventDefault();
        this.modalActive = false;
      },
    },
    computed:{
      isGitInfoShown(){
        if(GIT === undefined){
          return false
        }
        return GIT && GIT.BRANCH && !GIT.BRANCH.includes("master")
      }
    }
  }
</script>

<style lang='scss' scoped>
  body,
  html {
    width: 100%;
    height: 100%;
  }


  .lead {
    font-size: 18px;
    font-weight: 400;
  }

  .intro-header {
    padding-top: 50px; /* If you're making other pages, make sure there is 50px of padding to make sure the navbar doesn't overlap content! */
    padding-bottom: 50px;
    text-align: center;
    color: #f8f8f8;
    background: url(~assets/blur.png) no-repeat center center;
    background-size: cover;
  }

  .intro-logo > img {
    position: relative;
    max-width: 200px;

  }

  .intro-message {
    position: relative;
    padding-bottom: 20%;
  }

  .intro-message > h1 {
    margin: 0;
    text-shadow: 2px 2px 3px rgba(0, 0, 0, 0.6);
    font-size: 5em;
  }

  .intro-divider {
    width: 400px;
    border-top: 1px solid #f8f8f8;
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  }

  .intro-message > h3 {
    text-shadow: 2px 2px 3px rgba(0, 0, 0, 0.6);
  }

  @media(max-width: 767px) {
    .intro-message {
      padding-bottom: 15%;
    }

    .intro-message > h1 {
      font-size: 3em;
    }

    ul.intro-social-buttons > li {
      display: block;
      margin-bottom: 20px;
      padding: 0;
    }

    ul.intro-social-buttons > li:last-child {
      margin-bottom: 0;
    }

    .intro-divider {
      width: 100%;
    }
  }

  .network-name {
    text-transform: uppercase;
    font-size: 14px;
    font-weight: 400;
    letter-spacing: 2px;
  }

  .content-section-a {
    padding: 50px 0;
    background-color: #f8f8f8;
  }

  .content-section-b {
    padding: 50px 0;
    border-top: 1px solid #e7e7e7;
    border-bottom: 1px solid #e7e7e7;
  }

  .section-heading {
    margin-bottom: 30px;
  }

  .section-heading-spacer {
    float: left;
    width: 200px;
    border-top: 3px solid #e7e7e7;
  }

  .banner {
    padding: 100px 0;
    color: #f8f8f8;
    background: url(~assets/landing-page/banner-bg.jpg) no-repeat center center;
    background-size: cover;
  }

  .banner h2 {
    margin: 0;
    text-shadow: 2px 2px 3px rgba(0, 0, 0, 0.6);
    font-size: 3em;
  }

  .banner ul {
    margin-bottom: 0;
  }

  .banner-social-buttons {
    float: right;
    margin-top: 0;
  }

  @media(max-width: 1199px) {
    ul.banner-social-buttons {
      float: left;
      margin-top: 15px;
    }
  }

  @media(max-width: 767px) {
    .banner h2 {
      margin: 0;
      text-shadow: 2px 2px 3px rgba(0, 0, 0, 0.6);
      font-size: 3em;
    }

    ul.banner-social-buttons > li {
      display: block;
      margin-bottom: 20px;
      padding: 0;
    }

    ul.banner-social-buttons > li:last-child {
      margin-bottom: 0;
    }
  }

</style>
