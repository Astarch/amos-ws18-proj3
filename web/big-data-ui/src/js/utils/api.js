import axios from 'axios';
import qs from 'qs';
import store from 'store2'


/******************** Base API / Client Configuration ********************/

let port = "8081/"
let localhostUrl = "http://localhost:" + port;
let stagingUrl = "http://18.216.129.153:" + port;
let liveUrl = "http://18.216.122.218:" + port;
let baseUrl = stagingUrl
baseUrl = (GIT.BRANCH && !GIT.BRANCH.includes("master")) ? liveUrl : stagingUrl;



function writeAuthToken(token) {
    store.session('authToken', token)
    return token
}

function getAuthToken() {
    return store.session('authToken')
}


const axiosInstance = axios.create({
    baseURL: baseUrl,
});

if (getAuthToken() && getAuthToken().length > 5) {
    Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': getAuthToken() } });
}

// Intercepts all responses, retrieves the x auth token and sets it to all following requests headers!
axiosInstance.interceptors.response.use(response => {
    let authToken = getAuthToken()
    if (response.status == 401) {
        authToken = writeAuthToken("")
    } else if (response && response.headers && response.headers['x-auth-token']) {
        let authToken = writeAuthToken(response.headers['x-auth-token']);
        if (authToken && authToken.length > 5) {
            Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': authToken } });
        }
    }

    return response;
}, error => {
    // Do something with response error
    return Promise.reject(error);
});

export default axiosInstance;

export const isLoggedIn = () => {
    let authToken = getAuthToken() ? getAuthToken() : authToken;
    return authToken && authToken.length > 5
}

export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getData = (path) => axiosInstance.get(path);

const graph = {
    getData
};

const api = {
    auth,
    user,
    graph
}

export {
    api
};