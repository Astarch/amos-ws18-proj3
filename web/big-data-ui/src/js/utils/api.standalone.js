import axios from 'axios';
import qs from 'qs';
import storage from 'store2'
import store from 'src/js/store';

import trump from 'src/assets/wordcounts/trump'
const availableTerms = ["trump"]



/******************** Base API / Client Configuration ********************/

function writeAuthToken(token) {
  storage.set('authToken', token)
  return token
}

function getAuthToken() {
  return storage.get('authToken')
}

export const isLoggedIn = () => {
  let authToken = getAuthToken() ? getAuthToken() : authToken;
  return authToken && authToken.length > 5
}

export const logoutUser = () => {
  writeAuthToken("")
}


/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => {
  return new Promise((resolve, reject) => {
    if (username == "user2" && password == "pass2") {
      const response = {
        data: {
          id: 123,
          username: "user2",
          email: "user2@test.de",
          firstname: "John",
          lastname: "Doee",
          authorities: [],
          isLoggedIn: true
        }
      }
      writeAuthToken("loggedIn")
      resolve(response)
    } else {
      const errorResponse = {
        status: 401
      }
      reject(errorResponse)
    }
  })
}
//axiosInstance.post('/auth/login', qs.stringify({
//    username: username,
//    password: password
//}));

const postCompleteRegistration = (username,
  password,
  firstname = username,
  lastname = username,
  email = username + "@test.com",
  address = "addres1",
  phone = "phoneNr") => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "password": password,
//    "firstname": firstname,
//    "lastname": lastname,
//    "email": email,
//    "address": address,
//    "phone": phone
//});

const postRegistration = (username, firstname, lastname,
  email,
  password) => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "firstname": firstname,
//    "lastname": lastname,
//    "password": password,
//    "email": email,
//});



const auth = {
  postLogin,
  postRegistration,
  postCompleteRegistration
};

const localFirebaseUrl = "http://localhost:5000/pretrendr-d5939/us-central1"
const remoteFirebaseUrl = "https://us-central1-pretrendr-d5939.cloudfunctions.net"
const firebaseUrl = remoteFirebaseUrl
/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) => {
  /*return new Promise((resolve, reject)=> {
      if(query.indexOf("trump")>= 0){
          const response = {
              data: trump
          }
          resolve(response)
      }else{
          const errorResponse = {
              status: 404
          }
          reject(errorResponse)
      }
  })*/
  return axios.get(`${firebaseUrl}/getWordcountData?query=${query}&method=${method}`)

  //axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);
}
const uploadDataToFirestore = (query, method, timerange, data) => {
  const uploadData = {
    query,
    method,
    timerange,
    data,
  };
  return axios.post(`${firebaseUrl}/addWordcountData`, uploadData)    
}

const getCachedTrends = () => axios.get(`${firebaseUrl}/getAllCachedTerms`)


const graph = {
  getWordcountByDay,
  getCachedTrends
};

const api = {
  auth,
  graph
}

export {
  api
};
