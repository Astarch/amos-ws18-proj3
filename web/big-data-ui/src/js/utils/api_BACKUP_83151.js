import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

<<<<<<< HEAD
export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username, firstname,lastname,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "firstname": firstname,
    "lastname": lastname,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) =>
    axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);

const graph = {
    getWordcountByDay
};

const uploadDataToFirestore = (query, method, timerange, data) => {
    const uploadData = {
      query,
      method,
      timerange,
      data,
    };
    return axios.post('https://us-central1-pretrendr-d5939.cloudfunctions.net/addWordcountData', uploadData)    
  }

const api = {
    auth,
    user,
    graph,
    uploadDataToFirestore
}
=======
const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi
>>>>>>> initial commit to use frontend without backend

export {
    isLoggedIn,
    logoutUser,
    api
}
