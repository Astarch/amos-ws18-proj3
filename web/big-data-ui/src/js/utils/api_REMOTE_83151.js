import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi

export {
    isLoggedIn,
    logoutUser,
    api
}
