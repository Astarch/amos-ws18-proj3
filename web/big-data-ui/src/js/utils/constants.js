const FormTypeEnum = {"login": 1, "register": 2};
Object.freeze(FormTypeEnum);

export {
  FormTypeEnum
}
