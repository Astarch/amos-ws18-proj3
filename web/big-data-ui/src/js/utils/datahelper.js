function normalizeData(graphdata) {
    //console.time("normalizeData")
    let max = Object.values(graphdata).reduce((max, current) => {
        return current > max ? current : max
    }, -1)
    let normalizeFactor = 1 / (max / 100)
    let normalized = Object.keys(graphdata).reduce((previous, current) => {
        previous[current] = Math.round(graphdata[current] * normalizeFactor)
        return previous;
    }, {});
    //console.timeEnd("normalizeData")
    return normalized
}

function normalizeDataArray(dataArray) {
    let normalizedArray = dataArray.map(value => {
        return Object.assign({}, value, { data: normalizeData(value.data) })
    })
    return normalizedArray
}

function median(values) {
    values = values.slice(0).sort(function(a, b) { return a - b; });

    return middle(values);
}

function middle(values) {
    var len = values.length;
    var half = Math.floor(len / 2);

    if (len % 2)
        return (values[half - 1] + values[half]) / 2.0;
    else
        return values[half];
}

function calculateMinMaxAvgMedianOfData(graphdata) {
    const values = Object.values(graphdata)
    let min = parseInt(Math.min(...values), 10);
    let max = parseInt(Math.max(...values), 10);
    let sum = 0;
    for (var i = 0; i < values.length; i++) {
        sum += parseInt(values[i], 10); //don't forget to add the base
    }
    let avg = parseInt(sum / values.length, 10);
    let _median = parseInt(median(values), 10);

    return Object.assign({}, {
        min,
        max,
        avg,
        median: _median
    })
}

function aggregateDataArray(dataArray) {
    let additionalDataArray = dataArray.map(value => {
        return Object.assign({}, value, calculateMinMaxAvgMedianOfData(value.data))
    })
    return additionalDataArray
}



export {
    normalizeDataArray,
    aggregateDataArray
}