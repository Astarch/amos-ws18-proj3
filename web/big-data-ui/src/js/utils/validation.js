const emailRegex = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
const smallCharRegex = /[a-z]+/;
const capitalCharRegex = /[A-Z]+/;
const numberCharRegex = /[0-9]+/;
const specialCharRegex = /[^A-Za-z0-9]+/;


// calculates the strength of a password and returns the strength on a scale from
// 0 to 8 where 8 is a very strong password and 0 a very weak one
const getPasswordStrength = (password) => {
  let strength = 0;

  if (password.length > 0) {
    strength += smallCharRegex.test(password); // contains a-z
    strength += capitalCharRegex.test(password); // contains A-Z
    strength += numberCharRegex.test(password); // contains 0-9
    strength += specialCharRegex.test(password); // contains special character
    strength += Math.floor(Math.min(4, password.length / 3)); // 0-12 characters result in 0-4 points
  }
  return strength;
}

const isMailValid = (mail) => mail.length > 0 && emailRegex.test(mail);
const isPasswordValid = (password) => password.length > 4;
const isNameValid = (name) => name.length > 2;


export {
  isMailValid,
  isPasswordValid,
  isNameValid,
  getPasswordStrength
}

