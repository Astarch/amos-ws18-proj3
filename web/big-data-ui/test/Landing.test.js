import { shallow, mount } from 'vue-test-utils'
import Vue from 'vue'
import Landing from 'src/js/components/onboarding/Landing'

import {FormTypeEnum} from 'src/js/utils/constants';

describe('Landing.test.js', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(Landing, {
      data: { // Replace data value with this fake data
        modalActive: false,
        modalType: FormTypeEnum.login,
        requestAnswer: '',
        FormTypeEnum
      }
    })
  })

  it('equals modalActive to false', () => {
    expect(cmp.vm.modalActive).toBeFalsy()
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})
