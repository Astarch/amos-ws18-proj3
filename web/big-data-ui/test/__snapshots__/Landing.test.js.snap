// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing.test.js has the expected html structure 1`] = `
<div
  id="landing-page"
>
  <!---->
   
  <a
    name="about"
  />
   
  <div
    class="intro-header"
    style="padding-top: 50px;"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <div
            class="intro-logo"
          >
            <img
              class="img-fluid"
              src="~assets/logo/logo.png"
            />
          </div>
           
          <div
            class="intro-message"
          >
            <h1>
              pretrendr
            </h1>
             
            <h3>
              Analyse and Visualize your data and identify trends
            </h3>
             
            <hr
              class="intro-divider"
            />
             
            <ul
              class="list-inline intro-social-buttons"
            >
              <li>
                <a
                  class="btn btn-default btn-lg"
                  href="#register"
                >
                  Register Now
                </a>
              </li>
               
              <li>
                <a
                  class="btn btn-default btn-lg"
                  href="#login"
                >
                  Login
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <a
    name="services"
  />
   
  <div
    class="content-section-a"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-5 col-sm-6"
        >
          <hr
            class="section-heading-spacer"
          />
           
          <div
            class="clearfix"
          />
           
          <h2
            class="section-heading"
          >
            Death to the Stock Photo:
            <br />
            Special Thanks
          </h2>
           
          <p
            class="lead"
          >
            A special thanks to 
            <a
              href="http://join.deathtothestockphoto.com/"
              target="_blank"
            >
              Death to the Stock Photo
            </a>
            
            for providing the photographs that you see in this template. Visit their website to become a member.
          </p>
        </div>
         
        <div
          class="col-lg-5 col-lg-offset-2 col-sm-6"
        >
          <img
            alt=""
            class="img-responsive"
            src="~assets/landing-page/ipad.png"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="content-section-b"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-5 col-lg-offset-1 col-sm-push-6  col-sm-6"
        >
          <hr
            class="section-heading-spacer"
          />
           
          <div
            class="clearfix"
          />
           
          <h2
            class="section-heading"
          >
            3D Device Mockups
            <br />
            by PSDCovers
          </h2>
           
          <p
            class="lead"
          >
            
            Turn your 2D designs into high quality, 3D product shots in seconds using free Photoshop actions by 
            <a
              href="http://www.psdcovers.com/"
              target="_blank"
            >
              PSDCovers
            </a>
            ! Visit their website to download some of their awesome, free photoshop actions!
          
          </p>
        </div>
         
        <div
          class="col-lg-5 col-sm-pull-6  col-sm-6"
        >
          <img
            alt=""
            class="img-responsive"
            src="~assets/landing-page/dog.png"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="content-section-a"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-5 col-sm-6"
        >
          <hr
            class="section-heading-spacer"
          />
           
          <div
            class="clearfix"
          />
           
          <h2
            class="section-heading"
          >
            Google Web Fonts and
            <br />
            Font Awesome Icons
          </h2>
           
          <p
            class="lead"
          >
            This template features the 'Lato' font, part of the 
            <a
              href="http://www.google.com/fonts"
              target="_blank"
            >
              Google Web Font library
            </a>
            , as well as 
            <a
              href="http://fontawesome.io"
              target="_blank"
            >
              icons from Font Awesome
            </a>
            .
          </p>
        </div>
         
        <div
          class="col-lg-5 col-lg-offset-2 col-sm-6"
        >
          <img
            alt=""
            class="img-responsive"
            src="~assets/landing-page/phones.png"
          />
        </div>
      </div>
    </div>
  </div>
   
  <a
    name="contact"
  />
   
  <div
    class="banner"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-6"
        >
          <h2>
            Connect to pretrendr:
          </h2>
        </div>
         
        <div
          class="col-lg-6"
        >
          <ul
            class="list-inline banner-social-buttons"
          >
            <li>
              <a
                class="btn btn-default btn-lg"
                href="https://github.com/Astarch/amos-ws18-proj3"
              >
                <i
                  class="fa fa-github fa-fw"
                />
                 
                <span
                  class="network-name"
                >
                  Github
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
   
  <!---->
   
  <!---->
</div>
`;
